docs/docs/api-ref.md

```diff
@@ -5,257 +5,3010 @@ layout: docs
-self.server.sites.create(new_site)
+import tableauserverclient as TSC
+
+# create a auth request
+tableau_auth = TSC.TableauAuth('USERNAME', 'PASSWORD')
+
+# create an instance for your server
+server = TSC.Server('http://SERVER_URL')
+
+# call the sign-in method with the request
+server.auth.sign_in(tableau_auth)
+
 ```
 
-### Get Site by ID
 
-Gets the site with the given ID.
+**See Also**  
+[Sign in and Out](sign-in-out)  
+[Server](#server)
+
+<br>
+<br>
+
+
+#### auth.sign out
 
 ```py
-Sites.get_by_id(id)
+auth.sign_out()
 ```
+Signs you out of the current session.
 
-### Get Sites
+The method takes care of invalidating the authentiction token. For more information, see [Sign in and Out](sign-in-out).
 
-Gets the first 100 sites on the server. To get all the sites, use the Pager.
+REST API: [Sign Out](http://onlinehelp.tableau.com/current/api/rest_api/en-us/help.htm#REST/rest_api_ref.htm#Sign_Out%3FTocPath%3DAPI%2520Reference%7C_____78){:target="_blank"}
+
+**Example**
 
 ```py
-Sites.get()
+
+server.auth.sign_out()
+
+
 ```
 
-### Update Site
 
-Modifies a site. The site item object must include the site ID and overrides all other settings.
+
+
+**See Also**  
+[Sign in and Out](sign-in-out)  
+[Server](#server)
+
+<br>
+<br>
+
+
+
+
+## Connections
+
+The connections for Tableau Server data sources and workbooks are represented by a `ConnectionItem` class.  You can call data source and view methods to query or update the connection information.  The `ConnectionCredentials` class represents the connection information you can update. 
+
+### ConnectionItem class
 
 ```py
-Sites.update(site_item_object)
+ConnectionItem()
 ```
 
-### Delete Site
+**Attributes**  
+
+Name   |  Description     
+ :--- | : ---    
+`datasource_id` |  The identifier of the data source. 
+`datasource_name` |  The name of the data source.
+`id`  |  The identifer of the connection.
+`connection_type`  |  The type of connection. 
+
+
+Source file: models/connection_item.py  
+
+<br>
+<br>
+
 
-Deletes the site with the given ID.
+
+### ConnectionCredentials class
 
 ```py
-Sites.delete(id)
+ConnectionCredentials(name, password, embed=True, oauth=False)
 ```
 
-## Projects
 
-Source files: server/endpoint/projects_endpoint.py
+The `ConnectionCredentials` class corresponds to workbook and data source connections. 
+
+In the Tableau Server REST API, there are separate endopoints to query and update workbook and data source connections. 
+
+**Attributes**
+
+Attribute | Description
+:--- | :---
+`name`     | The username for the connection.
+`embed_password`  |  (Boolean) Determines whether to embed the passowrd (`True`) for the workbook or data source connection or not (`False`). 
+`password`  |  The password used for the connection.   
+`server_address`   |  The server address for the connection.   
+`server_port`   |  The port used by the server.  
```

Comment:
I updated "server_port" to say "the port used for the connection"

BTW I added some attributes to the ConnectionItem class (the username, password, etc.) that I missed earlier.  Also, I realized that the ConnectionCredentials is really just used when publishing workbooks and data sources. I didn't have that stated that way.